### PR TITLE
PMM-9243 custom folder and dashboard test

### PR DESCRIPTION
**Pull request by vyarosh**
* Includes fixes for dashboards folders test
* introduces new assert plugin due to 'assert' lib issue with crushing allure. see: https://www.screencast.com/t/dUQt02vW
* requires updating dependencies or installing manually: https://www.npmjs.com/package/codeceptjs-chai

### CODE REVIEW

File: tests/pages/api/grafanaAPI.js [hunk 1]
```diff
@@ -125,4 +127,50 @@ module.exports = {
...
+  },
+
+  /**
+   * Encapsulates multiple actions require do delete folder to keep test body short.
+   */
+  async safeDeletePmmFolder() {
+    const searchResultObj = await this.lookupFolderByName('PMM');
+
+    if (searchResultObj && Object.prototype.hasOwnProperty.call(searchResultObj, 'uid')) {
+      await this.deleteFolder(searchResultObj.uid);
```
Comment (puneet0191) [inline tests/pages/api/grafanaAPI.js:171]: I mentioned that PMM folder was not always empty, so why deleting it?
image

**Review by puneet0191 — CHANGES_REQUESTED** [review]

File: tests/pages/api/grafanaAPI.js (hunk 1 above)
Comment (vyarosh) [reply tests/pages/api/grafanaAPI.js:171]: removed tis strategy and added if instead.

**Comment by vyarosh** [comment]
https://pmm.cd.percona.com/blue/organizations/jenkins/pmm2-upgrade-tests/detail/pmm2-upgrade-tests/8067/pipeline

**Comment by vyarosh** [comment]
https://pmm.cd.percona.com/blue/organizations/jenkins/pmm2-ui-test-nightly/detail/pmm2-ui-test-nightly/829/pipeline

**Review by puneet0191 — APPROVED** [review]

**Pull request merged**